crates/ruff/src/linter.rs

```diff
@@ -50,26 +51,29 @@ impl<T> LinterResult<T> {
     }
 }
 
+pub type DiagnosticsAndImports = (Vec<Diagnostic>, FxHashMap<PathBuf, Vec<Import>>);
+pub type MessagesAndImports = (Vec<Message>, FxHashMap<PathBuf, Vec<Import>>);
 pub type FixTable = FxHashMap<&'static Rule, usize>;
 
 /// Generate `Diagnostic`s from the source code contents at the
 /// given `Path`.
 #[allow(clippy::too_many_arguments)]
-pub fn check_path(
-    path: &Path,
-    package: Option<&Path>,
-    contents: &str,
+pub fn check_path<'a>(
+    path: &'a Path,
+    package: Option<&'a Path>,
+    contents: &'a str,
     tokens: Vec<LexResult>,
-    locator: &Locator,
-    stylist: &Stylist,
-    indexer: &Indexer,
-    directives: &Directives,
-    settings: &Settings,
+    locator: &'a Locator,
+    stylist: &'a Stylist,
+    indexer: &'a Indexer,
+    directives: &'a Directives,
+    settings: &'a Settings,
```

Comment:
Can you explain why the lifetime annotation is necessary? We aren't using it in any return type.

Reply:
I think this is probably a hangover from when I was trying to make the imports work on a `&Path` rather than a `PathBuf`, so yes I will need to get rid of these